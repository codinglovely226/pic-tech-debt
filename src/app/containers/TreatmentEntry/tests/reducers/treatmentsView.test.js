/* eslint-disable */

import TreatmentsViewReducer from '../../reducers/treatmentsView';

import {
  resetTreatmentsView,
  updateTreatmentsViewContent,
  updateTreatmentsViewDisposition,
  updateTreatmentsViewTreatments,
} from '../../actions/case';

describe('TreatmentsViewReducer', () => {
  it('returns the initial state', () => {
    expect(TreatmentsViewReducer(undefined, {})).toMatchSnapshot();
  });
  it('reset treatments view', () => {
    expect(TreatmentsViewReducer({'content': 'Test Content', 'disposition': 'Test Disposition', 'spi': {'name': 'Test SPI'}, 'treatmentList': ['Test Item 1']}, resetTreatmentsView())).toMatchSnapshot();
  });
  it('update treatments view content', () => {
    expect(TreatmentsViewReducer({'content': 'Test Content', 'disposition': 'Test Disposition', 'spi': {'name': 'Test SPI'}, 'treatmentList': ['Test Item 1']}, updateTreatmentsViewContent('Test Content'))).toMatchSnapshot();
  });
  it('update treatments view disposition', () => {
    expect(TreatmentsViewReducer({'content': 'Test Content', 'disposition': 'Test Disposition', 'spi': {'name': 'Test SPI'}, 'treatmentList': ['Test Item 1']}, updateTreatmentsViewDisposition('Test Disposition'))).toMatchSnapshot();
  });
  it('update referrals view treatments', () => {
    expect(TreatmentsViewReducer({'content': 'Test Content', 'disposition': 'Test Disposition', 'spi': {'name': 'Test SPI'}, 'treatmentList': ['Test Item 1']}, updateTreatmentsViewTreatments(['Test Item 1', 'Test Item 2']))).toMatchSnapshot();
  });
});
