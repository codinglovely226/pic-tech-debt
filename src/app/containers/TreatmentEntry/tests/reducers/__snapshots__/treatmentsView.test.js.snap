exports[`TreatmentsViewReducer reset treatments view 1`] = `
Object {
  "content": "",
  "disposition": "",
  "spi": Object {
    "name": "Test SPI",
  },
  "treatmentList": Array [],
}
`;

exports[`TreatmentsViewReducer returns the initial state 1`] = `
Object {
  "content": "",
  "disposition": "",
  "spi": Object {
    "id": "",
    "job": "",
    "name": "",
    "org": "",
    "roles": Array [],
    "siteId": "",
  },
  "treatmentList": Array [],
}
`;

exports[`TreatmentsViewReducer update referrals view treatments 1`] = `
Object {
  "content": "Test Content",
  "disposition": "Test Disposition",
  "spi": Object {
    "name": "Test SPI",
  },
  "treatmentList": Array [
    "Test Item 1",
    "Test Item 2",
  ],
}
`;

exports[`TreatmentsViewReducer update treatments view content 1`] = `
Object {
  "content": "Test Content",
  "disposition": "Test Disposition",
  "spi": Object {
    "name": "Test SPI",
  },
  "treatmentList": Array [
    "Test Item 1",
  ],
}
`;

exports[`TreatmentsViewReducer update treatments view disposition 1`] = `
Object {
  "content": "Test Content",
  "disposition": "Test Disposition",
  "spi": Object {
    "name": "Test SPI",
  },
  "treatmentList": Array [
    "Test Item 1",
  ],
}
`;
